src/managers/RoleManager.js

```diff
@@ -123,6 +123,18 @@ class RoleManager extends BaseManager {
       });
   }
 
+  /**
+   * Gets the managed role a user created when joining the guild, if any
+   * <info>Only ever available for bots</info>
+   * @param {UserResolvable} user The user to access the bot role for
+   * @returns {?Role}
+   */
+  botRoleFor(user) {
+    user = this.client.users.resolve(user);
+    if (!user) return null;
+    return this.cache.find(role => role.tags && role.tags.botID === user.id) || null;
+  }
```

Comment:
```suggestion
  botRoleFor(user) {
    user = this.client.users.resolveID(user);
    return this.cache.find(role => role.tags && role.tags.botID === user) || null;
  }
```
This way there's not even a cached user object needed, since we only need the ID to find the bot role.

Reply:
implemented variation in b1fe25c to still keep the fast return if no input is provided (the resolveID call seems to be neglectable) and increase verbosity by not naming the id "user"